gpio: shell: Replace gpio listen with blink

DRuffer opened the pull request:
This is a frequently used command during hardware bringup
The listen command was not actually implemented
Also uncrustified and check_compliance.py

Signed-off-by: Dennis Ruffer <daruffer@gmail.com>

DRuffer committed:
gpio: shell: Replace gpio listen with blink

This is a frequently used command during hardware bringup
The listen command was not actually implemented
Also uncrustified and check_compliance.py

Signed-off-by: Dennis Ruffer <daruffer@gmail.com>

gmarull reviewed (approved):
LGTM. I think compliance check failure is a false positive, I can't see how MISRA Rule 5.7 is being violated here (cc: @ceolin)

carlescufi reviewed (approved):
LGTM, thank you!

jakub-uC commented on drivers/gpio/gpio_shell.c:
```diff
@@ -141,11 +144,53 @@ static int cmd_gpio_set(const struct shell *shell,
+			       SHELL_CMD(conf, NULL, "Configure GPIO", cmd_gpio_conf),
+			       SHELL_CMD(get, NULL, "Get GPIO value", cmd_gpio_get),
+			       SHELL_CMD(set, NULL, "Set GPIO", cmd_gpio_set),
+			       SHELL_CMD(blink, NULL, "Blink GPIO", cmd_gpio_blink),
```
You can use macro `SHELL_CMD_ARG` so the shell can check parameters count. You will not need to check it in the command handler.

DRuffer replied:
Done, but left the other parameter checks

DRuffer committed:
gpio: shell: Use macro SHELL_CMD_ARG to check parameters count

The args_number and args_no structures could be removed
But other parameter checks we still preserved.

Signed-off-by: Dennis Ruffer <daruffer@gmail.com>

DRuffer commented on drivers/gpio/gpio_shell.c:
```diff
@@ -141,11 +128,53 @@ static int cmd_gpio_set(const struct shell *shell,
+/* 500 msec = 1/2 sec */
+#define SLEEP_TIME_MS   500
+
+static int cmd_gpio_blink(const struct shell *shell,
```
I do not understand this error message: drivers/gpio/gpio_shell.c:134:WARNING: Violation to rule 5.7 (Tag name should be unique) tag: shell

The shell parameter is defined like this is every shell command I have seen.

How should it be defined to pass this rule?

DRuffer replied:
@gmarull claimed this was a "false positive", so can I get some help on getting this through the process?

gmarull replied:
@ceolin ^

henrikbrixandersen replied:
Change it to `const struct shell *sh` and the error will go away.

gmarull replied:
I can't see how `const struct shell *shell` violates MISRA 5.7 (just read the manual and tried with https://gimpel.com/demo.html).

ceolin replied:
MISRA is a little bit confuse about from my understanding. "A tag name shall be a unique identifier" and object is an identifier as stated in other rules, so it may be conflicting with tag name. Although https://gimpel.com/demo.html does not raise a violation, parasoft (that is the tool used by the project) does. Some of them can be seen here https://codescan.zephyrproject.org:8443/grs/dtp/explorers/violations?buildId=v2.7.0-rc1-48-g6e26512bd5&filterId=2&rule=MISRAC2012-RULE_5_7-a

```
struct multiboot_info multiboot_info;
...
time_t timeutil_timegm(const struct tm *tm)
```

I think the main problem is that MISRA-C does not list this case in this rule examples.

DRuffer replied:
Fixed!

ceolin replied:
@gmarull ^

gmarull replied:
Thanks for the clarification. MISRA is clearly vague here, to the point some checkers warn and others don't... :-)

DRuffer committed:
gpio: shell: Change shell *shell to shell *sh

To fix MISRA Rule 5.7 violation

Signed-off-by: Dennis Ruffer <daruffer@gmail.com>

DRuffer commented:
All checks have passed, so what's next?
